boxoffice/models/invoice.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+from decimal import Decimal
+from boxoffice.models import db, BaseMixin
+from baseframe import __
+from coaster.utils import LabeledEnum
+
+__all__ = ['Invoice', 'InvoiceLineItem']
+
+
+class Invoice(BaseMixin, db.Model):
+    __tablename__ = 'invoice'
+    __uuid_primary_key__ = True
+    __table_args__ = (db.UniqueConstraint('organization_id', 'invoice_no'),)
+
+    invoicee_name = db.Column(db.Unicode(255), nullable=True)
+    invoicee_email = db.Column(db.Unicode(254), nullable=True)
+    invoice_no = db.Column(db.Unicode(32), nullable=True)
+    invoiced_at = db.Column(db.DateTime, nullable=True)
+    street_address = db.Column(db.Unicode(255), nullable=True)
+    city = db.Column(db.Unicode(255), nullable=True)
+    state = db.Column(db.Unicode(255), nullable=True)
+    country = db.Column(db.Unicode(255), nullable=True)
+    postcode = db.Column(db.Unicode(8), nullable=True)
+    # GSTIN in the case of India
+    taxid = db.Column(db.Unicode(255), nullable=True)
+    # India specific: this is the state short code. Eg: KA for Karnataka
+    place_of_supply = db.Column(db.Unicode(3), nullable=True)
+
+    customer_order_id = db.Column(None, db.ForeignKey('customer_order.id'), nullable=False, index=True, unique=False)
+    order = db.relationship('Order', backref=db.backref('invoices', cascade='all, delete-orphan'))
+
+    organization_id = db.Column(None, db.ForeignKey('organization.id'), nullable=False)
+    organization = db.relationship('Organization', backref=db.backref('invoices', cascade='all, delete-orphan', lazy='dynamic'))
```

Comment:
If orders also belong to an organization, this is denormalizing it.  You'll need a column validator to ensure this field can never point to another organization. (I can understand denormalization as an indexing strategy, but this sort of thing can result in pretty bad data integrity errors.)